share/completions/git.fish

```diff
@@ -1355,17 +1369,56 @@ complete -f -c git -n '__fish_git_using_command gc' -l keep-largest-pack -d 'Ign
+complete -f -c git -n '__fish_git_using_command grep' -s E -l extended-regexp -d 'Use POSIX extended regexp for patterns'
+complete -f -c git -n '__fish_git_using_command grep' -s G -l basic-regexp -d 'Use POSIX basic regexp for patterns'
+complete -f -c git -n '__fish_git_using_command grep' -s P -l perl-regexp -d 'Use Perl-compatible regular expressions for patterns. Git must be compiled with support for this'
+complete -f -c git -n '__fish_git_using_command grep' -s F -l fixed-strings -d 'Don’t interpret pattern as a regex'
+complete -f -c git -n '__fish_git_using_command grep' -s n -l line-number -d 'Prefix the line number to matching lines'
+complete -f -c git -n '__fish_git_using_command grep' -l column -d 'Prefix the 1-indexed byte-offset of the first match from the start of the matching line'
+complete -f -c git -n '__fish_git_using_command grep' -s l -l files-with-matches -d 'Show only the names of files that contain matches'
+complete -f -c git -n '__fish_git_using_command grep' -s L -l files-without-match -d 'Show only the names of files that do not contain matches'
+complete -f -c git -n '__fish_git_using_command grep' -s z -l null -d 'Use \\0 as the delimiter for pathnames in the output, and print them verbatim'
+complete -f -c git -n '__fish_git_using_command grep' -s o -l only-matching -d 'Print only the matched parts of a matching line'
+complete -f -c git -n '__fish_git_using_command grep' -s c -l count -d 'Instead of showing every matched line, show the number of lines that match'
+complete -f -c git -n '__fish_git_using_command grep' -l no-color -d 'Turn off match highlighting, even when the configuration file gives the default to color output'
+complete -f -c git -n '__fish_git_using_command grep' -l break -d 'Print an empty line between matches from different files'
+complete -f -c git -n '__fish_git_using_command grep' -l heading -d 'Show the filename above the matches in that file instead of at the start of each shown line'
+complete -f -c git -n '__fish_git_using_command grep' -s p -l show-function -d 'Show the preceding line that contains the function name of the match, unless the matching line is a function name itself'
+complete -f -c git -n '__fish_git_using_command grep' -s W -l function-context -d 'Show the surrounding text from the previous line containing a function name up to the one before the next function name'
+complete -f -c git -n '__fish_git_using_command grep' -s e -d 'The next parameter is the pattern'
+complete -f -c git -n '__fish_git_using_command grep' -l and -d 'Combine patterns using and'
+complete -f -c git -n '__fish_git_using_command grep' -l or -d 'Combine patterns using or'
+complete -f -c git -n '__fish_git_using_command grep' -l not -d 'Combine patterns using not'
+complete -f -c git -n '__fish_git_using_command grep' -l all-match -d 'Limit the match to files that have lines to match all of them when giving multiple pattern expressions combined with --or'
```

Comment:
This and some other descriptions are quite long, maybe we can give a shorter summary